examples/datasets/plot_limo_data.py

```diff
@@ -111,68 +169,248 @@
-# patterns evoked by the presented face-stimuli.
-# Here, one would expect that faces with high phase-coherence evoke a stronger
-# response, as participants should be better at identifying these faces.
+# As shown above, there are some considerable differences between the
+# activation patterns evoked by stimuli with low vs. high phase-coherence.
 #
-# Create design matrix for linear regression. We'll use the information
-# contained in the ``limo_epochs.metadata``.
-design = limo_epochs.metadata.copy()
-design = design.assign(intercept=1)  # add intercept
+# In particular, differences appear to be most pronounced along fronto-central
+# and occipital electrodes.
+
+###############################################################################
+# Prepare data for linear regression analysis
+# --------------------------------------------
+#
+# Next, we can test the significance of these differences using linear
+# regression.But before we go on, we'll start by interpolating any missing
+# channels in the LIMO epochs structure. Some subjects in the datasets contain
+# missing channels (stored in ``limo_epochs.info[‘bads’]``) as these were
+# dropped during preprocessing of the data.
+
+limo_epochs.interpolate_bads(reset_bads=True)
+
+# Furthermore, we'll drop the EOG channels (marked by the "EXG" prefix)
+# present in the LIMO epochs structure.
+limo_epochs.drop_channels(['EXG1', 'EXG2', 'EXG3', 'EXG4'])
+
+###############################################################################
+# Define predictor variables and design matrix
+# --------------------------------------------
+#
+# First, we need to create a design matrix containing information about the
+# variables (i.e., predictors) we want to use for prediction of brain
+# activity patterns. For this purpose, we'll use the information contained in
+# the ``limo_epochs.metadata``. Here, we'll explore the effect of
+# phase-coherence as well as the effect of Face A vs. Face B.
+
+# name of predictors + intercept
+predictor_vars = ['face a - face b', 'phase-coherence', 'intercept']
+
+# create design matrix
+design = limo_epochs.metadata[['phase-coherence', 'face']]
 design['face a - face b'] = np.where(design['face'] == 'A', 1, -1)
-names = ['intercept', 'face a - face b', 'phase-coherence']
+design['intercept'] = 1
+design = design[predictor_vars]
+
+###############################################################################
+# Now we can set up the linear model to be used in the analysis.
+# For the purpose of demonstration, we'll use scikit-learn's LinearRegression
+# (see :class:`sklearn.linear_model.LinearRegression`) in this example.
+#
+# Basically, we're creating a wrapper, ``STLinearRegression``, for
+# sklearn's ``LinearRegression``, which computes a
+# `least squares`_ solution for our data given the provided design matrix. The
+# and results are stored within ``LinearRegression`` object for convenience.
+
+
+class STLinearRegression(LinearRegression):
```

Comment:
this seems *way* too complicated, and doesn't obviously do anything that the subsequent call to `mne.stats.linear_regression` doesn't also do.  Unless there's a compelling reason to show both (i.e., they do different things or yield different results) I'd cut this (more complex) model, and edit the surrounding prose accordingly.

Reply:
This is a larger discussion we partially wanted to kick off here.

In the long run, we want a more powerful functionality than `mne.stats.linear_regression`. This probably means a class anyways. How can we proceed here to shed technical debt and position us well for more interesting models?

Reply:
Right, the idea behind defining the a linear regression class in the example was to provide some ground for discussion of some issues that came up during GSoC (see #6710) and show how a linear regression "object" could look like in the near future. Thus, while it was intended for the class defined here to yield the same results as `mne.stats.linear_regression, it wasn't necessarily intended to be present in the final version of the example. 

At the moment, `mne.stats.linear_regression` computes a lot of stuff (t-values, p-values, etc.) and puts them into a dict of evokeds. In my opinion, we don’t really need that a single subject level. So one idea was to define a class which basically brings the data in form to interface with sklearn and sets up the linear model. Then, if we want to, we could just add methods that emulate the output of `mne.stats.linear_regression`. Alternatively, we could just let the class output what we need on a subject level (e.g., the „raw“ beta coefficients or maybe event the predictions) while keeping everything tidy for later manipulation (e.g., compute t-values, p-vales, on a group-level).

The main advantage of having a linear regression "object", is that it increases the flexibility for manipulation of the linear model and its results, for instance, if we want to add weights to the model, or we want to visualize the single subject results or are mainly interested in moving on to group-level analysis.

So I guess the main question is whether you guys think this is the right way to go. What are your thoughts on this?

Reply:
I understand the advantages of a class-based approach, and I agree that if you're seeking to expand MNE-Python's capabilities for linear modeling then wrappers around sklearn classes is a sensible way to go.  But the title of this example is "single trial linear regression analysis with the LIMO dataset", which (as you demonstrate) can be achieved with the existing `mne.stats.linear_regression` function. That said, to expand on my comments on [my PR into this PR](https://github.com/JoseAlanis/mne-python/pull/1#discussion_r338339894), I'd be happy if you showed the `mne.stats.linear_regression` model first, and then said something like:

> what if we want to do something more complicated, like XXXXX. To do that, we can use `sklearn`, but we'll create a thin wrapper around it's `LinearRegression` class so that it returns evoked objects.

etc.  So my vote is still either (1) cut the class-based example, or (2) modify the class-based example so that it demonstrates some functionality that is not currently possible with our own stats code. If you can do that in say ~30 lines of code in a section at the end of the example, I'd be okay with it.  If it takes much more than that, perhaps a second example that focuses solely on custom regression models is warranted.

Reply:
Ok, thanks for the feedback. Let’s go with Nr. 1. I'll drop the class-based approach and open a new PR that focuses exclusively on that issue. Seems like the most efficient way to go. I'll let you know when the code is ready for the final review.